packages/web-components/fast-foundation/src/date-picker/date-picker.ts

```diff
@@ -0,0 +1,1259 @@
+        const meridians: { text: string; action: () => void }[] = [
+            {
+                text: amText,
+                action: this.handleMeridianClicked.bind(this, "AM"),
+            },
+            {
+                text: pmText,
+                action: this.handleMeridianClicked.bind(this, "PM"),
+            },
+        ];
+        const meridian: string =
+            this.selectedDate?.meridian || (new Date().getHours() >= 12 ? "PM" : "AM");
+        if (meridian === "PM") {
+            meridians.reverse();
+        }
+
+        return {
+            hours,
+            minutes,
+            meridians,
+        };
+    }
+
+    /**
+     * Helper function used to create a matrix as a multi-dimensional array from a flat array
+     * @param array - a flat array to build into a multi-demensional array
+     * @param itemsPerRow - the maximum numer of items in each array row
+     * @returns
+     */
+    public arrayToMatrix(array: any[], itemsPerRow: number): any[][] {
+        return array.reduce((matrix, item, index) => {
+            if (index % itemsPerRow === 0) {
+                matrix.push([item]);
+            } else {
+                matrix[matrix.length - 1].push(item);
+            }
+            return matrix;
+        }, []);
+    }
+
+    /**
+     * Creates an array of selectable months for the month picker
+     * @returns an array of month labels
+     * @public
+     */
+    public getMonths(): {
+        action: () => void;
+        selected: boolean;
+        text: string;
+    }[] {
+        return new Array(12).fill(null).map((_, index) => {
+            return {
+                action: () => this.handleMonthClicked(index + 1, this.monthView),
+                keyup: this.handleMonthKeyup.bind(this, index + 1, this.monthView),
+                selected:
+                    this.calendarMonth === index + 1 &&
+                    this.calendarYear === this.yearView,
+                text: this.dateFormatter.getMonth(index + 1, "short"),
+            };
+        });
+    }
+
+    /**
+     * Creates an array of selectable years for the year picker
+     * @returns an array of years starting from this.yearRange
+     * @public
+     */
+    public getYears(): {
+        action: () => void;
+        text: string;
+    }[] {
+        return new Array(12).fill(null).map((_, index) => {
+            const year = this.yearView + index;
+            const text = this.dateFormatter.getYear(year);
+            return {
+                action: () => this.handleYearClicked(year),
+                keyup: this.handleYearKeyup.bind(this, year),
+                text,
+            };
+        });
+    }
+
+    /**
+     * Boolean to indicate if the user is interacting with the date-picker
+     * @public
+     */
+    public overFlyout: boolean = false;
+
+    /**
+     * Boolean to indicate the flyout is open
+     * @public
+     */
+    @observable
+    public flyoutOpen: boolean = false;
```

Comment:
The [combobox](https://github.com/microsoft/fast/blob/575c89b2e98afbf91b08c1e18f2a080a0b779052/packages/web-components/fast-foundation/src/combobox/combobox.ts#L129) and [select](https://github.com/microsoft/fast/blob/575c89b2e98afbf91b08c1e18f2a080a0b779052/packages/web-components/fast-foundation/src/select/select.ts#L46) use an attribute name open and the [tooltip](https://github.com/microsoft/fast/blob/575c89b2e98afbf91b08c1e18f2a080a0b779052/packages/web-components/fast-foundation/src/tooltip/tooltip.ts#L30) uses an attribute named visible.

Is there a reason to be a property only and not use one of the existing names, ie "open"? (We are working on a menu-button which is essentially a picker and wondered if there was an established convention).